File: leetcode/0477.go
package leetcode

func totalHammingDistance(nums []int) int {
	n, ans := len(nums), 0
	for i := 0; i < 30; i++ {
		c := 0
		for _, v := range nums {
			if (v>>i)&1 == 1 {
				c++
			}
		}
		ans += c * (n - c)
	}
	return ans
}

File: leetcode/0477.cpp
class Solution {
public:
    int totalHammingDistance(vector<int>& nums) {
        int n = nums.size(), ans = 0;
        for (int i = 0; i < 30; i++) {
            int c = 0;
            for (auto& each : nums)
                if ((each >> i) & 1)
                    ++c;
            ans += c * (n - c);
        }
        return ans;
    }
};